invokeai/backend/model_manager/load/model_loaders/stable_diffusion.py

```diff
@@ -71,46 +81,61 @@ def _load_model(
+                ModelVariantType.Normal: StableDiffusionPipeline,
+                ModelVariantType.Inpaint: StableDiffusionInpaintPipeline,
+            },
+            BaseModelType.StableDiffusionXL: {
+                ModelVariantType.Normal: StableDiffusionXLPipeline,
+                ModelVariantType.Inpaint: StableDiffusionXLInpaintPipeline,
+            },
+        }
         assert isinstance(config, MainCheckpointConfig)
-        base = config.base
-
+        try:
+            load_class = load_classes[config.base][config.variant]
+        except KeyError as e:
+            raise Exception(f"No diffusers pipeline known for base={config.base}, variant={config.variant}") from e
+        original_config_file = self._app_config.legacy_conf_path / config.config_path
         prediction_type = config.prediction_type.value
         upcast_attention = config.upcast_attention
-        image_size = (
-            1024
-            if base == BaseModelType.StableDiffusionXL
-            else 768
-            if config.prediction_type == SchedulerPredictionType.VPrediction and base == BaseModelType.StableDiffusion2
-            else 512
-        )
-
-        self._logger.info(f"Converting {model_path} to diffusers format")
-
-        loaded_model = convert_ckpt_to_diffusers(
-            model_path,
-            output_path,
-            model_type=self.model_base_to_model_type[base],
-            original_config_file=self._app_config.legacy_conf_path / config.config_path,
-            extract_ema=True,
-            from_safetensors=model_path.suffix == ".safetensors",
-            precision=self._torch_dtype,
-            prediction_type=prediction_type,
-            image_size=image_size,
-            upcast_attention=upcast_attention,
-            load_safety_checker=False,
-            num_in_channels=VARIANT_TO_IN_CHANNEL_MAP[config.variant],
-        )
-        return loaded_model
+
+        # Without SilenceWarnings we get log messages like this:
+        # site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
+        # warnings.warn(
+        # Some weights of the model checkpoint were not used when initializing CLIPTextModel:
+        # ['text_model.embeddings.position_ids']
+        # Some weights of the model checkpoint were not used when initializing CLIPTextModelWithProjection:
+        # ['text_model.embeddings.position_ids']
+
+        with SilenceWarnings():
+            pipeline = load_class.from_single_file(
+                config.path,
+                config=original_config_file,
+                torch_dtype=self._torch_dtype,
+                local_files_only=True,
+                prediction_type=prediction_type,
+                upcast_attention=upcast_attention,
+                load_safety_checker=False,
+            )
```

Comment:
As I was stepping through this function to try to figure out the source of the reproducibility issue, I noticed that we are passing in a `config` parameter, which is silently ignored in `from_single_file(...)`.

It does accept an `original_config_file` param, which seemed like what we were trying to pass. But, I got an error when I updated the call to be:
```suggestion
            pipeline = load_class.from_single_file(
                config.path,
                original_config_file=original_config_file,
                torch_dtype=self._torch_dtype,
                local_files_only=True,
                prediction_type=prediction_type,
                upcast_attention=upcast_attention,
                load_safety_checker=False,
            )
```

Error:
```bash
[2024-06-20 21:26:20,255]::[InvokeAI]::ERROR --> Error while invoking session 64f480c7-64c2-499c-b57c-eba61d92408b, invocation c3ae055d-ee79-43d2-9e3c-2800b7d098af (compel): StableDiffusionPipeline.__init__() missing 2 required positional arguments: 'text_encoder' and 'tokenizer'
[2024-06-20 21:26:20,255]::[InvokeAI]::ERROR --> Traceback (most recent call last):
  File "/home/ryanjdick3/src/InvokeAI/invokeai/app/services/session_processor/session_processor_default.py", line 129, in run_node
    output = invocation.invoke_internal(context=context, services=self._services)
  File "/home/ryanjdick3/src/InvokeAI/invokeai/app/invocations/baseinvocation.py", line 289, in invoke_internal
    output = self.invoke(context)
  File "/home/ryanjdick3/.pyenv/versions/3.10.14/envs/InvokeAI_3.10.14/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/ryanjdick3/src/InvokeAI/invokeai/app/invocations/compel.py", line 67, in invoke
    tokenizer_info = context.models.load(self.clip.tokenizer)
  File "/home/ryanjdick3/src/InvokeAI/invokeai/app/services/shared/invocation_context.py", line 369, in load
    return self._services.model_manager.load.load_model(model, _submodel_type)
  File "/home/ryanjdick3/src/InvokeAI/invokeai/app/services/model_load/model_load_default.py", line 71, in load_model
    ).load_model(model_config, submodel_type)
  File "/home/ryanjdick3/src/InvokeAI/invokeai/backend/model_manager/load/load_default.py", line 56, in load_model
    locker = self._load_and_cache(model_config, submodel_type)
  File "/home/ryanjdick3/src/InvokeAI/invokeai/backend/model_manager/load/load_default.py", line 75, in _load_and_cache
    loaded_model = self._load_model(config, submodel_type)
  File "/home/ryanjdick3/src/InvokeAI/invokeai/backend/model_manager/load/model_loaders/stable_diffusion.py", line 58, in _load_model
    return self._load_from_singlefile(config, submodel_type)
  File "/home/ryanjdick3/src/InvokeAI/invokeai/backend/model_manager/load/model_loaders/stable_diffusion.py", line 121, in _load_from_singlefile
    pipeline = load_class.from_single_file(
  File "/home/ryanjdick3/.pyenv/versions/3.10.14/envs/InvokeAI_3.10.14/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
    return fn(*args, **kwargs)
  File "/home/ryanjdick3/.pyenv/versions/3.10.14/envs/InvokeAI_3.10.14/lib/python3.10/site-packages/diffusers/loaders/single_file.py", line 313, in from_single_file
    pipe = pipeline_class(**init_kwargs)
TypeError: StableDiffusionPipeline.__init__() missing 2 required positional arguments: 'text_encoder' and 'tokenizer'
```

@lstein I'm hoping you can shed some light on the history of our `$INVOKEAI_ROOT/configs/` files, and why they differ from the [originals that diffusers uses](https://github.com/huggingface/diffusers/blame/b69fd990ad8026f21893499ab396d969b62bb8cc/src/diffusers/loaders/single_file_utils.py#L55-L62).

Reply:
The 1.5-era config files are the ones distributed with the original latent diffusion model codebase. They've never been updated and haven't seemed to cause any problems. It seems that the parameters to `load_single_file()` are inconsistent. In diffusers 0.27, `AutoencoderKL.load_single_file()` accepts a parameter of `config`, but `StableDiffusionPipeline.load_single_file()` accepts `original_config` (which says it does the same thing). In diffusers 0.29, the StableDiffusion classes have two parameters, `original_config` and `config`, which do different things.

I'm just removing the `config` and `original_config` arguments and letting the classes download their own config files by setting `local_files_only=False`. While playing around with this I discovered that `load_single_file()` requires a series of auxiliary modules (the same ones which used to be located in `models/core`) that are cached to the hub, so we need to allow for dynamic downloads anyway.